### PR TITLE
Reserve a subnet equal to network pool

### PR DESCRIPTION
**Pull request by davidefalcone1**
This PR adds the capability to reserve a network which is equal to a network pool.

### CODE REVIEW

File: pkg/liqonet/ipam.go [hunk 1]
```diff
@@ -89,8 +89,20 @@ func (liqoIPAM *IPAM) AcquireReservedSubnet(reservedNetwork string) error {
 		return err
 	}
 	if ok && reservedNetwork == pool {
-		klog.Errorf("Network %s is equal to a pool", reservedNetwork)
-		return fmt.Errorf("network %s is equal to a pool so it cannot be reserved", reservedNetwork)
+		klog.Infof("Network %s is equal to a pool, acquiring first half..", reservedNetwork)
+		mask := getMask(reservedNetwork)
```
Comment (palexster) [inline pkg/liqonet/ipam.go:93]: Can you please briefly comment on this logic?

Comment (davidefalcone1) [reply pkg/liqonet/ipam.go:96]: Sure!

**Review by palexster — APPROVED** [review]

File: pkg/liqonet/ipam.go [hunk 2]
```diff
@@ -251,7 +266,7 @@ func (liqoIPAM *IPAM) mapNetwork(network string) (string, error) {
 
 /* Helper function to get a mask from a net.IPNet */
 func getMask(network string) uint8 {
-	stringMask := network[len(network)-2:]
+	stringMask := strings.Split(network, "/")[1]
```
Comment (alacuku) [inline pkg/liqonet/ipam.go:269]: can we avoid manipulating strings in order to get the mask length? i would use the "net" package to achieve the same result. something like
```
	_, network, err := net.ParseCIDR(network)
	if err != nil{
		return err
	}
	ones := network.Mask.Size())
	return ones

Comment (davidefalcone1) [reply pkg/liqonet/ipam.go:269]: Good idea

**Review by alacuku — APPROVED** [review]

**Pull request merged**